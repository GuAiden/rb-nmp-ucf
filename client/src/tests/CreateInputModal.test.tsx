import Enzyme, { shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import CreateInputModal from '../components/Forms/InputForm/createinputmodal';

Enzyme.configure({ adapter: new Adapter() });

it('InputModal renders without failure', () => {
  const addInput = jest.fn();
  const component = shallow(<CreateInputModal onAddInput={addInput} />);
  expect(component).toMatchSnapshot();
});

it('should display a blank login form, with conversion and output checked by default', () => {
  const onAddInputMock = jest.fn();
  const wrapper = shallow(<CreateInputModal onAddInput={onAddInputMock} />);
  wrapper.find('Button.modal-add-wrapper').simulate('click');
  expect(onAddInputMock).toHaveBeenCalled();
  expect(onAddInputMock).toMatchInlineSnapshot(`
    [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "conversion": false,
            "output": false,
          },
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  `);
});
