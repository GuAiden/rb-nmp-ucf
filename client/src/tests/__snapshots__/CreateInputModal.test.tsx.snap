// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputModal renders without failure 1`] = `ReactWrapper {}`;
